pkg/rpc/server_helper_test.go

```diff
@@ -0,0 +1,182 @@
+	Result  result.Version `json:"result"`
+	ID      int            `json:"id"`
+}
+
+// IntResultResponse struct for testing.
+type IntResultResponse struct {
+	Jsonrpc string `json:"jsonrpc"`
+	Result  int    `json:"result"`
+	ID      int    `json:"id"`
+}
+
+// StringResultResponse struct for testing.
+type StringResultResponse struct {
+	Jsonrpc string `json:"jsonrpc"`
+	Result  string `json:"result"`
+	ID      int    `json:"id"`
+}
+
+// GetBlockResponse struct for testing.
+type GetBlockResponse struct {
+	Jsonrpc string `json:"jsonrpc"`
+	Result  struct {
+		Version           int    `json:"version"`
+		Previousblockhash string `json:"previousblockhash"`
+		Merkleroot        string `json:"merkleroot"`
+		Time              int    `json:"time"`
+		Height            int    `json:"height"`
+		Nonce             int    `json:"nonce"`
+		NextConsensus     string `json:"next_consensus"`
+		Script            struct {
+			Invocation   string `json:"invocation"`
+			Verification string `json:"verification"`
+		} `json:"script"`
+		Tx []struct {
+			Type       string      `json:"type"`
+			Version    int         `json:"version"`
+			Attributes interface{} `json:"attributes"`
+			Vin        interface{} `json:"vin"`
+			Vout       interface{} `json:"vout"`
+			Scripts    interface{} `json:"scripts"`
+		} `json:"tx"`
+		Confirmations int    `json:"confirmations"`
+		Nextblockhash string `json:"nextblockhash"`
+		Hash          string `json:"hash"`
+	} `json:"result"`
+	ID int `json:"id"`
+}
+
+// GetAssetResponse struct for testing.
+type GetAssetResponse struct {
+	Jsonrpc string `json:"jsonrpc"`
+	Result  struct {
+		AssetID    string `json:"assetID"`
+		AssetType  int    `json:"assetType"`
+		Name       string `json:"name"`
+		Amount     string `json:"amount"`
+		Available  string `json:"available"`
+		Precision  int    `json:"precision"`
+		Fee        int    `json:"fee"`
+		Address    string `json:"address"`
+		Owner      string `json:"owner"`
+		Admin      string `json:"admin"`
+		Issuer     string `json:"issuer"`
+		Expiration int    `json:"expiration"`
+		IsFrozen   bool   `json:"is_frozen"`
+	} `json:"result"`
+	ID int `json:"id"`
+}
+
+func initServerWithInMemoryChain(ctx context.Context, t *testing.T) (*core.Blockchain, http.HandlerFunc) {
+	net := config.ModeUnitTestNet
+	configPath := "../../config"
+	cfg, err := config.Load(configPath, net)
+	require.NoError(t, err, "could not load config")
+
+	memoryStore := storage.NewMemoryStore()
+	chain, err := core.NewBlockchain(memoryStore, cfg.ProtocolConfiguration)
+	require.NoError(t, err, "could not create chain")
+
+	go chain.Run(ctx)
+	initBlocks(t, chain)
+
+	serverConfig := network.NewServerConfig(cfg)
+	server := network.NewServer(serverConfig, chain)
+	rpcServer := NewServer(chain, cfg.ApplicationConfiguration.RPCPort, server)
+	handler := http.HandlerFunc(rpcServer.requestHandler)
+
+	return chain, handler
+}
+
+func initBlocks(t *testing.T, chain *core.Blockchain) {
```

Comment:
Why did you choose to throw away the chain we had in LevelDB?

Reply:
We talked about it in person.
Anyway will leave here some explanation: there we had over 10 000 blocks for some reason which is not required for unit testing and other thing which looks more important for me is to have in test controllable instances of blocks/tx/accounts so we're able to create them from actual code(which is under the test), put them into the chain and check that through rpc call they could be reached.